### PR TITLE
DeveloperGuide: update converting docs to PDF format

### PR DESCRIPTION
**Pull request by eugenepeh**
Ready for review

### CODE REVIEW

**Comment by eugenepeh** [comment]
@damithc as requested in gitter

**Comment by Zhiyuan-Amos** [comment]
This overlaps with #673 right, and there seems to be conflict in the way you are deciding on how to generate the PDF files :P

**Comment by eugenepeh** [comment]
@Zhiyuan-Amos yeah xD this comes after #673

**Comment by eugenepeh** [comment]
ready for review

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -443,13 +443,13 @@ We use https://www.google.com/chrome/browser/desktop/[Google Chrome] for convert
 
 Here are the steps to convert the project documentation files to PDF format.
 
-.  Make sure you have set up GitHub Pages as described in <<DeveloperGuide#publishing-documentation, Publishing Documentation>>.
-.  Using Chrome, go to the URL of the documentation file. e.g. For <<UserGuide#, UserGuide.adoc>>, the URL will be `\https://<your-username-or-organization-name>.github.io/addressbook-level4/docs/UserGuide.html`.
-.  Click on the `Print` option in Chrome's menu.
+.  Make sure you have followed the <<UsingGradle#rendering-asciidoc-files, UsingGradle.adoc>> instructions to convert the AsciiDoc files in `docs` to HTML format.
```
Comment (pyokagan) [inline docs/DeveloperGuide.adoc]: ``to convert the AsciiDoc files in `docs` to HTML format`` -> ``to convert the AsciiDoc files in the `docs/` directory to HTML format``

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -443,13 +443,13 @@ We use https://www.google.com/chrome/browser/desktop/[Google Chrome] for convert
...
+.  Go to your generated HTML files in the `build/docs` folder, right click on them and select `Open with -> Google Chrome`.
```
Comment (pyokagan) [inline docs/DeveloperGuide.adoc]: ``and select `Open with -> Google Chrome`.`` -> ``and select `Open with` -> `Google Chrome`.``

**Comment by pyokagan** [comment]
> ```
> DeveloperGuide: modify converting docs instructions to use local file
> ```

Please add some simple background info on why the docs instructions need to be updated as well. (e.g. `It is easier for developers to just generate the PDF directly on their local machines without having to go through the steps of publishing the documentation on GitHub Pages.`)

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -443,9 +443,9 @@ We use https://www.google.com/chrome/browser/desktop/[Google Chrome] for convert
 
 Here are the steps to convert the project documentation files to PDF format.
 
-.  Make sure you have set up GitHub Pages as described in <<DeveloperGuide#publishing-documentation, Publishing Documentation>>.
-.  Using Chrome, go to the URL of the documentation file. e.g. For <<UserGuide#, UserGuide.adoc>>, the URL will be `\https://<your-username-or-organization-name>.github.io/addressbook-level4/docs/UserGuide.html`.
-.  Click on the `Print` option in Chrome's menu.
+.  Make sure you have followed the <<UsingGradle#rendering-asciidoc-files, UsingGradle.adoc>> instructions to convert the AsciiDoc files in the `docs/` directory to HTML format
```
Comment (Zhiyuan-Amos) [inline docs/DeveloperGuide.adoc]: missing period :P

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -443,13 +443,13 @@ We use https://www.google.com/chrome/browser/desktop/[Google Chrome] for convert
...
```
Comment (pyokagan) [inline docs/DeveloperGuide.adoc]: Sorry, one last nit: `Make sure you have followed the <<UsingGradle#rendering-asciidoc-files, UsingGradle.adoc>> instructions` -> `Follow the <<UsingGradle#rendering-asciidoc-files, UsingGradle.adoc>> instructions`. Previously, you only needed to set up GitHub Pages once. Now, it is best to always run `gradle asciidoctor` whenever you wish to convert to PDF to ensure that you have the latest generated files.

Comment (eugenepeh) [reply docs/DeveloperGuide.adoc]: oh yes, that's right. Didn't realize that part, thanks :D

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (eugenepeh) [reply docs/DeveloperGuide.adoc]: ouch... need to feedback github to add in `commit --amend --no-edit` feature for such things 😄

Comment (Zhiyuan-Amos) [reply docs/DeveloperGuide.adoc]: the workflow of having to commit, rebase and squash is... :(

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -443,13 +443,13 @@ We use https://www.google.com/chrome/browser/desktop/[Google Chrome] for convert
 
 Here are the steps to convert the project documentation files to PDF format.
 
-.  Make sure you have set up GitHub Pages as described in <<DeveloperGuide#publishing-documentation, Publishing Documentation>>.
-.  Using Chrome, go to the URL of the documentation file. e.g. For <<UserGuide#, UserGuide.adoc>>, the URL will be `\https://<your-username-or-organization-name>.github.io/addressbook-level4/docs/UserGuide.html`.
-.  Click on the `Print` option in Chrome's menu.
+.  Follow the <<UsingGradle#rendering-asciidoc-files, UsingGradle.adoc>> instructions to convert the AsciiDoc files in the `docs/` directory to HTML format.
```
Comment (pyokagan) [inline docs/DeveloperGuide.adoc]: Oh whoops, it should be `Follow the instructions in UsingGradle.adoc`...

**Comment by Zhiyuan-Amos** [comment]
> For commit titles for both commits, capitalise the first word after the colon e.g DeveloperGuide: Modify...

You missed this, but it's ok this time (just a small thing), but take note in future ^^

**Comment by pyokagan** [comment]
> For commit titles for both commits, capitalise the first word after the colon e.g DeveloperGuide: Modify...

@Zhiyuan-Amos The official guide actually recommends it to be lower case :-P See the examples: https://oss-generic.github.io/process/docs/FormatsAndConventions.html#commit-message

**Comment by Zhiyuan-Amos** [comment]
Gosh opps. I deserved to be sacked :(

**Comment by eugenepeh** [comment]
@Zhiyuan-Amos it's okay, i've remembered I read somewhere to lower case but can't remember where :P
don't try to get yourself sacked on purpose 😄 we still need you in addressbook

**Pull request merged**